### PR TITLE
fix: ensure flagset set validation is called

### PR DESCRIPTION
**Pull request by shaneutt**
**What this PR does / why we need it**:

Previously the wrong Set() method was called when updating
values from ENV vars, this patch ensures that each flag runs
it's local Set() method to ensure any validation there is run.

**Notes**

When bad ENV values are provided, an error like this will occur:

```shell
ERROR: could not set CONTROLLER_CONTROLLER_UDPINGRESS value from environment to false: "false" is not a valid EnablementStatus
exit status 25
```

**Which issue this PR fixes**

Fixes https://github.com/Kong/kubernetes-ingress-controller/issues/1498

### CODE REVIEW

File: railgun/cmd/rootcmd/env.go [hunk 1]
```diff
@@ -22,7 +29,10 @@ func bindEnvVars(cmd *cobra.Command, _ []string) {
 		}
 
 		if envValue, envSet := os.LookupEnv(envKey); envSet {
-			cmd.Flags().Set(f.Name, envValue)
+			if err := f.Value.Set(envValue); err != nil {
+				fmt.Fprintf(os.Stderr, "ERROR: could not set %s value from environment to %s: %s\n", envKey, envValue, err)
+				os.Exit(ExitCodeBadFlagValue)
```
Comment (ccfishk) [inline railgun/cmd/rootcmd/env.go:34]: Suggest to return error and let the caller decide whether continue or exit from invalid setting.

The function is to trying to abstract the validation process, but it acts on behalf of caller - exit program, which is not the validation's job. a little bit conflict.

Comment (ccfishk) [reply railgun/cmd/rootcmd/env.go:34]: Please return an error in 
https://github.com/Kong/kubernetes-ingress-controller/pull/1511/files#diff-4f4502a06cf38a24c5b34f902871497909c6c3794343ca49daeec5bc737d6ac2L16
for the failure in 
https://github.com/Kong/kubernetes-ingress-controller/pull/1511/files#diff-4f4502a06cf38a24c5b34f902871497909c6c3794343ca49daeec5bc737d6ac2R33
 and let the caller return from the program.

Comment (mflendrich) [reply railgun/cmd/rootcmd/env.go:34]: I second this suggestion. :+1:

File: railgun/cmd/rootcmd/env.go [hunk 2]
```diff
@@ -22,7 +29,10 @@ func bindEnvVars(cmd *cobra.Command, _ []string) {
 		}
 
 		if envValue, envSet := os.LookupEnv(envKey); envSet {
-			cmd.Flags().Set(f.Name, envValue)
+			if err := f.Value.Set(envValue); err != nil {
+				fmt.Fprintf(os.Stderr, "ERROR: could not set %s value from environment to %s: %s\n", envKey, envValue, err)
```
Comment (mflendrich) [inline railgun/cmd/rootcmd/env.go:33]: This log print may be unnecessary after this function returns an error (as suggested in the comment on the line below), but I think that we should use the logger here and in other situations like this.

Printing directly to stderr adds a new unrelated channel of reporting, disregards the logging configuration (level, format) and structure.

File: railgun/cmd/rootcmd/env.go [hunk 3]
```diff
@@ -22,7 +29,10 @@ func bindEnvVars(cmd *cobra.Command, _ []string) {
 		}
 
 		if envValue, envSet := os.LookupEnv(envKey); envSet {
-			cmd.Flags().Set(f.Name, envValue)
+			if err := f.Value.Set(envValue); err != nil {
```
Comment (mflendrich) [inline railgun/cmd/rootcmd/env.go:32]: Can we unit test this?

**Review by mflendrich — COMMENTED** [review]
Thank you for investigating the issue and finding the real root cause, triggered by the misconfiguration that I caused :bow:

File: railgun/cmd/rootcmd/env.go (hunk 1 above)
Comment (shaneutt) [reply railgun/cmd/rootcmd/env.go]: Sounds good, 1455d231067c3007b6dd26b3563f70ec5238912d resolves this :+1:

File: railgun/cmd/rootcmd/env.go (hunk 2 above)
Comment (shaneutt) [reply railgun/cmd/rootcmd/env.go]: gone with 1455d231067c3007b6dd26b3563f70ec5238912d  :+1:

File: railgun/cmd/rootcmd/env.go (hunk 3 above)
Comment (shaneutt) [reply railgun/cmd/rootcmd/env.go:26]: There are some unit tests that already exist for this, what do you have in mind?

Comment (shaneutt) [reply railgun/cmd/rootcmd/env.go:26]: a test has been added in 9b96ca359f39861e2660750f9a8eea41bb9ce9e7 :+1:

File: railgun/cmd/rootcmd/env.go [hunk 4]
```diff
@@ -22,7 +23,22 @@ func bindEnvVars(cmd *cobra.Command, _ []string) {
...
+
+	var wrappedErr error
+	if len(errs) > 0 {
+		for i, err := range errs {
+			if i == 0 {
+				wrappedErr = err
+			} else {
+				wrappedErr = fmt.Errorf("%v: %w", wrappedErr, err)
+			}
+		}
```
Comment (mflendrich) [inline railgun/cmd/rootcmd/env.go:40]: this error assembling functionality occupies more than 50% of this function and is untested.

Please consider doing one of the following:
- just `return err` on the first error in this function
- implement the "wrap all errors in one **über**error" functionality as a util function (probably outside of this package) and test it

Comment (shaneutt) [reply railgun/cmd/rootcmd/env.go]: Ok, changed to just return a single error in 34a05c004c2dcee8be690fed215dbd0afcac4654 :+1:

**Review by mflendrich — APPROVED** [review]
Following an out-of-band discussion I've pushed a commit that changes the behavior from capturing the last error to capturing the first error and bailing out. Due to the interface of `VisitAll` this is only possible by means of `panic`/`recover`.

LGTM

**Review by rainest — APPROVED** [review]

**Pull request merged**